### PR TITLE
Update Motif Testing & Speed up alignments

### PR DESCRIPTION
**Pull request by blackpan2**
_This is a branch off of `motif-maker` which is up for PR in #103 and is BLOCKED till that is merged. The base branch will be updated to dev after 103 is merged._

Goal:
- Updates to Motif Testing
  - Use `MotifFile` object where appropriate
- Speed up alignments:
  - On `dev` we are passing around a `results` hashmap between functions. By doing so we are increasing the time of alignment requests by ~20 seconds.
  - By eliminating streams we were able to speed up the alignment request as well

Dev Change:
- Caching has been moved to the production profile

### CODE REVIEW

**Comment by jmiller656** [comment]
unblocked because #103 was merged

File: src/main/java/org/moltimate/moltimatebackend/controller/ControllerExceptionHandler.java [hunk 1]
```diff
@@ -38,4 +38,9 @@
     protected ResponseEntity<Object> invalidFile(InvalidFileException ex, WebRequest request) {
         return handleExceptionInternal(ex, ex.getMessage(), new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
     }
+
+    @ExceptionHandler(value = {MotifFileParseException.class})
```
Comment (blackpan2) [inline src/main/java/org/moltimate/moltimatebackend/controller/ControllerExceptionHandler.java]: This is only used in `FileUtils : readMotifFile`when it is not a `InvalidFileException`.

This gives us a generic runtime response message for having difficulty parsing a file.

File: src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java [hunk 2]
```diff
@@ -64,97 +64,68 @@ public ActiveSiteAlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest a
         Page<Motif> motifs = motifService.queryByEcNumber(motifEcNumberFilter, pageNumber);
 
         log.info(String.format("Aligning active sites of %d PDB entries with %d motifs & %d custom motifs.",
-                sourceStructures.size(), motifs.getTotalElements(), customMotifs.size()));
+                sourceStructures.size(), motifs.getTotalElements(), customMotifFileList.size()));
 
         // Align structures with motifs from the database
         while (motifs.hasContent()) {
-            motifs.stream()
```
Comment (blackpan2) [inline src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java]: The stream loop here was bloating the length of an alignment request significantly (~30 seconds)

File: src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java [hunk 3]
```diff
@@ -64,97 +64,68 @@ public ActiveSiteAlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest a
...
 
         // Align structures with custom uploaded motifs
-        customMotifs.forEach((customMotif, customMotifStructure) -> {
-            alignActiveSiteStructureList(results, customMotif, customMotifStructure, sourceStructures, precision);
-        });
-
-        results.forEach((pdbId, alignments) -> results.replace(pdbId, alignments.stream()
-                .filter(Objects::nonNull)
-                .collect(Collectors.toList())));
+        if (customMotifFileList.size() > 0) {
```
Comment (blackpan2) [inline src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java]: By including this if we eliminate a for loop across a large number of entities (we would iterate through and do nothing with the interior stream) when we do not have to.

**Comment by steplica** [comment]
Great attention to detail. There are a lot of changes to core parts of the code so I trust you've tested the changes.
LGTM 👍

**Pull request merged**